Displays/PIXIE.txt

```diff
@@ -0,0 +1,117 @@
+       at address B. See below for a description of video ram.
+       If B is 0, the screen is disconnected.
+    1: MEM_MAP_FONT
+       Reads the B register, and maps the font ram to DCPU-16 ram starting
+       at address B. See below for a description of font ram.
+       If B is 0, the default font is used instead.
+    2: MEM_MAP_PALETTE
+       Reads the B register, and maps the palette ram to DCPU-16 ram starting
+       at address B. See below for a description of palette ram.
+       If B is 0, the default palette is used instead.
+    3: SET_BORDER_COLOR
+       Reads the B register, and sets the border color to palette index B&0xF
+    4: MEM_DUMP_FONT
+       Reads the B register, and writes the default font data to DCPU-16 ram
+       starting at address B.
+       Halts the DCPU-16 for 256 cycles
+    5: MEM_DUMP_PALETTE
+       Reads the B register, and writes the default palette data to DCPU-16
+       ram starting at address B.       
+       Halts the DCPU-16 for 16 cycles
+
+    16: SET_VIDEO_MODE
+       Reads the B register, and changes to specified video mode:
+
+          0: Text mode (LEM1802 standard)
+          1: Graphics mode, 2 colors
+          2: Graphics mode, 4 colors
+          3: Graphics mode, 8 colors
+          4: Graphics mode, 16 colors
+
+       Graphics mode will use the same video_ram starting address as text mode,
+       but video_ram size and layout are different. Graphics mode will also use
+       the same palette as text mode, but only the colors 0-n (depending on the
+       color depth) are used. Fontram is not relevant in graphics mode.
+
+
+Graphics mode video ram:
+    a 128 by 96 screen is organized in 1-4 bitplanes of 768 words each.
+    Each bitplane consists of 96 rows of 8 words. Number of bitplanes
+    depends on the number of colors. 2 color mode has only one bitplane,
+    while 16 color mode has four bitplanes.
+
+              number  memory  total
+    bitplane  colors  offset  memory
+    --------  ------  ------  ------
+       0         2        0     768   2, 4, 8 and 16 color modes
+       1         4      768    1536   4, 8 and 16 color modes
+       2         8     1536    2304   8 and 16 color modes
+       3        16     2304    3072   16 color mode
+
+    Bitplane number is used as a bit position to refer the color palette.
+    The bitplane 0 bit is the least significant bit in LEM palette. 
+        e.g. if bitplane 0 is clear, but bitplanes 1-3 are set,
+             the palette color would be 0b1110 -> 0xE -> 14
+
+    Gaphics are stored to bitplane in a linear fashion; each word of data
+    on a line is located at an address that is one greater than the word on
+    its left. i.e.  Each line is a "plus one" continuation of the previous
+    line.
+                   _______________________________________
+                  |    |    |    |    |    |    |    |    |
+                  | 20 | 21 | 22 | 23 | 24 | 25 | 26 | 27 |
```

Comment:
Is there some reason this bitplane is starting at 20, whereas the table above lists 0 as the base offset for the first bitplane?  (If you are showing this as "here's how it'd look if your MEM_MAP_SCREEN was 20", maybe make more explicit?